src/python/pants/backend/python/goals/tailor.py

```diff
@@ -126,15 +132,25 @@ async def find_putative_targets(
             all_pipenv_lockfile_files,
             all_pyproject_toml_contents,
         ) = await MultiGet(
-            Get(Paths, PathGlobs, req.path_globs("*requirements*.txt")),
-            Get(Paths, PathGlobs, req.path_globs("Pipfile.lock")),
+            Get(DigestContents, PathGlobs, req.path_globs("*requirements*.txt")),
+            Get(DigestContents, PathGlobs, req.path_globs("Pipfile.lock")),
             Get(DigestContents, PathGlobs, req.path_globs("pyproject.toml")),
         )
 
-        def add_req_targets(files: Iterable[str], alias: str, target_name: str) -> None:
-            unowned_files = set(files) - set(all_owned_sources)
+        def add_req_targets(files: Iterable[FileContent], alias: str, target_name: str) -> None:
+            contents = {i.path: i.content for i in files}
+            unowned_files = set(contents) - set(all_owned_sources)
             for fp in unowned_files:
                 path, name = os.path.split(fp)
+
+                if not validate(fp, contents[fp], alias):
+                    logger.warning(
+                        f"No target generated for `{fp}`. You'll need to create "
+                        "targets for its contents manually."
+                    )
+                    logger.warning("")
```

Comment:
Is this to have a newline? We have no precedent for this, and imo it's confusing to see `WARN` twice. 

If you really wanted a newline, it's better to use `\n` in the call above. But again, no precedent for that.

Reply:
Also, probably better to combine the warning with the implementation-specific warnings. We try to have logs be self-contained